### PR TITLE
chore: remove legacy interface function, RandomizedParams, which is no longer used

### PR DESCRIPTION
**Pull request by colin-axner**
## Description

While reviewing changes, I noticed this function had been removed from the SDK AppModuleSimulation function

closes: #3148 

### Commit Message / Changelog Entry

```bash
chore: remove legacy interface function `RandomizedParams`, which is no longer used.
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/10-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [x] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [x] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [x] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [x] Re-reviewed `Files changed` in the Github PR explorer.
- [x] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -59,6 +59,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 * (07-tendermint) [\#1896](https://github.com/cosmos/ibc-go/pull/1896) Remove error return from `IterateConsensusStateAscending` in `07-tendermint`.
 * (apps/27-interchain-accounts) [\#2638](https://github.com/cosmos/ibc-go/pull/2638) Interchain accounts host and controller Keepers now expects a keeper which fulfills the expected `exported.ScopedKeeper` interface for the capability keeper.
 * (06-solomachine) [\#2761](https://github.com/cosmos/ibc-go/pull/2761) Removed deprecated `ClientId` field from `Misbehaviour` and `allow_update_after_proposal` field from `ClientState`.
+* (apps) [#3149](https://github.com/cosmos/ibc-go/pull/3149) Remove legacy interface function, RandomizedParams, which is no longer used.
```
Comment (colin-axner) [inline CHANGELOG.md:62]:
```suggestion
* (apps) [#3149](https://github.com/cosmos/ibc-go/pull/3149) Remove legacy interface function `RandomizedParams`, which is no longer used.
```

File: modules/core/module.go [hunk 2]
```diff
@@ -175,11 +174,6 @@ func (AppModule) ProposalContents(_ module.SimulationState) []simtypes.WeightedP
 	return nil
```
Comment (julienrbrt) [inline modules/core/module.go:174]: You can delete this of all modules if you don't use it.
And as mentioned [here](https://github.com/cosmos/ibc-go/pull/3107#discussion_r1097611389), if you keep it, it should be still returning `WeightedProposalContent`.

Comment (colin-axner) [reply modules/core/module.go:174]: We will delete it as it is unused :+1: Thanks for reviewing!

**Review by damiannolan — APPROVED** [review]
🙏

**Pull request merged**